### PR TITLE
Add EL9 distro

### PR DESCRIPTION
**Pull request by szepeviktor**
@tillkruss Is this what you want?

### CODE REVIEW

File: build/fpm/fpm.sh [hunk 1]
```diff
@@ -67,6 +67,24 @@ packages=(
...
+  "rpm multi.el9  x86_64 7.4 20190902 $baseurl-php7.4-centos8-x86-64.tar.gz"
+
+  "rpm single.el9 aarch64 8.2 20210902 $baseurl-php8.2-centos8-aarch64.tar.gz"
+  "rpm single.el9 aarch64 8.1 20210902 $baseurl-php8.1-centos8-aarch64.tar.gz"
+  "rpm single.el9 aarch64 8.0 20200930 $baseurl-php8.0-centos8-aarch64.tar.gz"
+  "rpm single.el9 aarch64 7.4 20190902 $baseurl-php7.4-centos8-aarch64.tar.gz"
+  "rpm multi.el9  aarch64 8.2 20210902 $baseurl-php8.2-centos8-aarch64.tar.gz"
+  "rpm multi.el9  aarch64 8.1 20210902 $baseurl-php8.1-centos8-aarch64.tar.gz"
+  "rpm multi.el9  aarch64 8.0 20200930 $baseurl-php8.0-centos8-aarch64.tar.gz"
+  "rpm multi.el9  aarch64 7.4 20190902 $baseurl-php7.4-centos8-aarch64.tar.gz"
```
Comment (tillkruss) [inline build/fpm/fpm.sh]: These are all `el9` and not `centos8`.

File: rpm/el9/repodata/repomd.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<repomd xmlns="http://linux.duke.edu/metadata/repo" xmlns:rpm="http://linux.duke.edu/metadata/rpm">
+ <revision>1677169356</revision>
+</repomd>
```
Comment (tillkruss) [inline rpm/el9/repodata/repomd.xml]: You can delete this, it will be generated.

File: rpm/el9/repodata/repomd.xml.asc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+YPhu47qm9eFRulBGLVbGCF9lvYWK4Bm4R29y7yVcsMg6cnNVou+GBqlpt2QJfIiw
+hzfwjeTvg6t0QR6nIaf1LqqlS1zfrOerpWslnU0e8A14TwccuookxqjHiD3MEW9z
+Km+fG63H6bD8bNgZiQj22K8Jm3C/X4syup72Y2TpTOtEap7hdYAEwIy6VxjwEBOR
+KTYV9g8SdC4mKNx7wyNq2Clrnv3FCHSikMw+Nkt0zLv4vXghrpjXhwrO6ZGgZciz
+dgWHWxm4iyRC1CTcJYVRquQE+I2/ihPOIJGanPXW3sG0khxLsQMfnYs64HgvlFAs
+SjIa0GfxSwnLNl+uvu+cZ9kGfqOSIAw5coTdl+ZdiEfWGOJrG4HBg71OcOvr20AJ
+X+uji9L/EqfcAB9IvF5JrJj+6vp4WALc635lXGl2wFCDs9gy4rxu8bo60+0Ryfmr
+ucOUmkUIUmYzE2huRnRAoQiptZszklZooyFJtf87i467PhBbAyU=
+=KfQq
+-----END PGP SIGNATURE-----
```
Comment (tillkruss) [inline rpm/el9/repodata/repomd.xml.asc]: You can delete this, it will be generated.

File: build/fpm/src/rpm/config.single.el9.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "libzstd"
+    "lz4"
+    "php(api) = $php_api-64"
+    "php-msgpack"
+    "php-igbinary"
+)
+
+fpm_args=(
+  "--after-install /root/fpm/src/rpm/after-install.sh"
+)
```
Comment (tillkruss) [inline build/fpm/src/rpm/config.single.el9.sh:28]: Can you post a REMI EL9 package and we can compare them?

**Review by tillkruss — CHANGES_REQUESTED** [review]
Whoa my dude, let's go.

File: build/fpm/src/rpm/config.single.el9.sh (hunk 4 above)
Comment (szepeviktor) [reply build/fpm/src/rpm/config.single.el9.sh:28]: image

This is the directory structure. I think we don't have to change anything.
https://rpms.remirepo.net/enterprise/9/php80/x86_64/repoview/php-pecl-mcrypt.html

File: rpm/el9/repodata/repomd.xml (hunk 2 above)
Comment (szepeviktor) [reply rpm/el9/repodata/repomd.xml]: Done.

File: build/fpm/fpm.sh (hunk 1 above)
Comment (szepeviktor) [reply build/fpm/fpm.sh]: Replaced.

File: build/fpm/src/rpm/config.single.el9.sh (hunk 4 above)
Comment (tillkruss) [reply build/fpm/src/rpm/config.single.el9.sh:28]: Can you upload the contents of the `*.cpio` from another El9 module, so I can look at the dependencies, etc? Idealy the `redis` php82 one.

Comment (szepeviktor) [reply build/fpm/src/rpm/config.single.el9.sh:28]: Here you go 🐎 
https://github.com/szepeviktor/php82-php-pecl-redis5.el9.remi.x86_64

File: build/fpm/src/rpm/config.multi.el9.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+pkg_name="php$php_version_short-php-relay"
+pkg_provides="php$php_version_short-php-relay"
+pkg_identifier="el9"
+
+pkg_binary_dest=(
+    "opt/remi/php$php_version_short/root/usr/lib64/php/modules"
```
Comment (tillkruss) [inline build/fpm/src/rpm/config.multi.el9.sh:8]: This seems unchanged.

File: build/fpm/src/rpm/config.multi.el9.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+pkg_config_dest=(
+    "etc/opt/remi/php$php_version_short/php.d/60-redis.ini"
```
Comment (tillkruss) [inline build/fpm/src/rpm/config.multi.el9.sh]: This should be `60-relay.ini`, not `redis`.

File: build/fpm/src/rpm/config.multi.el9.sh [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+pkg_depends=(
+    "openssl"
+    "libzstd"
+    "lz4"
+    "php$php_version_short-php(api) = $php_api-64"
+    "php$php_version_short-php-msgpack"
+    "php$php_version_short-php-igbinary"
```
Comment (tillkruss) [inline build/fpm/src/rpm/config.multi.el9.sh:21]: Can you post the dependencies of `php82-php-pecl-redis5.el9.remi.x86_64`?

File: build/fpm/src/rpm/config.single.el9.sh [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+pkg_binary_dest=(
+    "usr/lib64/php/modules"
+    "usr/lib64/php-zts/modules"
+)
+
+pkg_config_dest=(
+    "etc/php.d/60-relay.ini"
+    "etc/php-zts.d/60-relay.ini"
+)
```
Comment (tillkruss) [inline build/fpm/src/rpm/config.single.el9.sh:15]: Thanks, the `php82-php-pecl-redis5.el9.remi.x86_64` was very helpful. Can you post one of Remi's "single" package as well so we can verify these haven't changed?

**Comment by szepeviktor** [comment]
viktor was out of order, not ready for hard work.

---

What needs to be done here?

**Pull request merged**